app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb

```diff
@@ -9,6 +9,25 @@ def parse
     targets = []
 
     case ems_event[:event_type]
+    when /^image\.delete/
+      targets << InventoryRefresh::Target.new(
+        :association => :miq_templates,
+        :manager_ref => {:ems_ref => ems_event[:vm_ems_ref]},
+        :manager_id  => ems_event.ext_management_system.id,
+        :event_id    => ems_event.id
+      )
+    when /^image/
+      if ems_event[:message].include?("successfully imported into image catalog")
+        targets << ems_event.ext_management_system
+      end
+    when /^network/
+      network_manager = ManageIQ::Providers::IbmCloud::PowerVirtualServers::NetworkManager.find_by(:parent_ems_id => ems_event[:ems_id])
```

Comment:
`ems_event` should be an EmsEvent object, which `ext_management_system` is already a loaded association for, so you should be able to do `ems_event.ext_management_system.network_manager` without doing the query manually.

Reply:
Nice! I'm glad to get rid of the queries.